design/expected-hardware-configuration-validation.md

```diff
@@ -0,0 +1,262 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# Hardware Configuration Validation
+
+## Table of Contents
+
+<!--ts-->
+   * [Hardware Configuration Validation](#title)
+      * [Table of Contents](#table-of-contents)
+      * [Summary](#summary)
+      * [Motivation](#motivation)
+         * [Goals](#goals)
+      * [Proposal](#proposal)
+         * [Implementation Details/Notes/Constraints ](#implementation-detailsnotesconstraints-optional)
+         * [Risks and Mitigations](#risks-and-mitigations)
+      * [Design Details](#design-details)
+         * [Work Items](#work-items)
+         * [Dependencies](#dependencies)
+         * [Test Plan](#test-plan)
+      * [References](#references)
+
+## Summary
+
+Code with the ability to perform the comparison on inspected and ready hosts and update the host with labels containing matched profiles.
+
+## Motivation
+
+We need to validate and compare hardware configuration against host hardware details-section.
+
+### Goals
+
+Purpose is to add a new CRD `HardwareClassificationController` to hold the  hardware details that reconciles those by looking for host CRs that matches with hardware classification profile and add label as "BMO-OK" for matched profile into baremetal host baremetalHost_crd.yaml.
+
+## Proposal
+
+We are creating proposal based on the discussion with metal3 community on issue #351:
+https://github.com/metal3-io/baremetal-operator/issues/351
+
+We compared introspection data of Ironic with metal3 schema and we found that default  hardware
+configuration needs to be added by introducing a new CRD `HardwareClassificationController`.
+
+We will write a CRD that checks inspected baremetal hosts against hardware configuration and add label as "BMO-OK" for matched profile into baremetal host baremetalHost_crd.yaml.
+  
+### Implementation Details/Notes/Constraints
```

Comment:
Generally for this section, I don't think we need to include the level of detail about which commands to run or what files will be created. We can stay at a higher level, and talk about creating the controller. We should be specific about the fields in the type, since that's the API we're defining.

Reply:
ok